cdap-adapters/src/main/java/co/cask/cdap/conversion/app/StreamConversionAdapter.java

```diff
@@ -16,17 +16,51 @@
 
 package co.cask.cdap.conversion.app;
 
-import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.FileSetProperties;
+import co.cask.cdap.api.schedule.Schedules;
+import co.cask.cdap.api.templates.AdapterConfigurer;
+import co.cask.cdap.api.templates.ApplicationTemplate;
+import com.google.gson.Gson;
+import org.apache.avro.mapreduce.AvroKeyInputFormat;
+import org.apache.avro.mapreduce.AvroKeyOutputFormat;
 
 /**
  * Application that converts a stream into a partitioned file set.
  */
-public class StreamConversionAdapter extends AbstractApplication {
+public class StreamConversionAdapter extends ApplicationTemplate<AdapterArgs> {
 
   @Override
   public void configure() {
     setDescription("Periodically reads stream events and writes them to a time partitioned fileset");
     addMapReduce(new StreamConversionMapReduce());
     addWorkflow(new StreamConversionWorkflow());
+
+    Schema eventSchema = Schema.recordOf(
+      "streamEvent",
+      Schema.Field.of("ts", Schema.of(Schema.Type.LONG)),
+      Schema.Field.of("header1", Schema.unionOf(Schema.of(Schema.Type.NULL), Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("header2", Schema.unionOf(Schema.of(Schema.Type.NULL), Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("ticker", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("num", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE)));
+    createDataset("converted", "timePartitionedFileSet", FileSetProperties.builder()
+      .setBasePath("converted")
+      .setInputFormat(AvroKeyInputFormat.class)
+      .setOutputFormat(AvroKeyOutputFormat.class)
+      .setEnableExploreOnCreate(true)
+      .setSerDe("org.apache.hadoop.hive.serde2.avro.AvroSerDe")
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", eventSchema.toString())
+      .build());
   }
+
+  @Override
+  public void configureAdapter(String adapterName, AdapterArgs args,
+                               AdapterConfigurer configurer) throws Exception {
+    configurer.addRuntimeArgument("adapter.args", new Gson().toJson(args));
+    configurer.setSchedule(Schedules.createTimeSchedule("test", "adapter schedule", "* * * * *"));
```

Comment:
We had discussed that the schedule will be added on start and not while configuring adapters. this shouldn't be there right?